### PR TITLE
Add package pykickstart version 3.36

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add package pykickstart version 3.36

###### Change Log  

- NA

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

Buddy Build
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=323577&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=322967&view=results)

### CODE REVIEW

File: SPECS-EXTENDED/pykickstart/pykickstart.spec [hunk 1]
```diff
@@ -0,0 +1,804 @@
...
+%bcond_with runtests
+Summary:        Python utilities for manipulating kickstart files.
+Name:           pykickstart
+Version:        3.36
+Release:        2%{?dist}
+License:        GPLv2 AND MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://fedoraproject.org/wiki/pykickstart
+# This is a Red Hat maintained package which is specific to
```
Comment (mfrw) [inline SPECS-EXTENDED/pykickstart/pykickstart.spec:13]: Probably we should double check if this comment is valid, as we have source0 pointing to github

File: SPECS-EXTENDED/pykickstart/pykickstart.spec [hunk 2]
```diff
@@ -0,0 +1,804 @@
...
+%package -n python3-kickstart
+Summary:        Python 3 library for manipulating kickstart files.
+Requires:       python3-requests
+
+%description -n python3-kickstart
+Python 3 library for manipulating kickstart files.  The binaries are found in
+the pykickstart package.
+
+%prep
+%setup -q -n %{name}-r%{version}
```
Comment (mfrw) [inline SPECS-EXTENDED/pykickstart/pykickstart.spec:43]:
```suggestion
%autosetup -n %{name}-r%{version}
```

File: SPECS-EXTENDED/pykickstart/pykickstart.spec [hunk 3]
```diff
@@ -0,0 +1,804 @@
...
+
+%build
+make PYTHON=python3
+
+%install
+make PYTHON=python3 DESTDIR=%{buildroot} install
+
+%check
+%if %{with runtests}
```
Comment (mfrw) [inline SPECS-EXTENDED/pykickstart/pykickstart.spec:52]: We should remove this conditional

File: SPECS-EXTENDED/pykickstart/pykickstart.spec [hunk 4]
```diff
@@ -0,0 +1,804 @@
...
+Source1:        %{name}-%{version}.tar.gz.asc
+BuildRequires:  gettext
+BuildRequires:  make
+BuildRequires:  python3-devel
+BuildRequires:  python3-requests
+BuildRequires:  python3-setuptools
+Requires:       python3-kickstart = %{version}-%{release}
+BuildArch:      noarch
+# Only required when building with runtests
+%if %{with runtests}
```
Comment (mfrw) [inline SPECS-EXTENDED/pykickstart/pykickstart.spec:26]:
```suggestion
%if %{with_check}
```

**Review by mfrw — COMMENTED** [review]
A few suggestions

File: SPECS-EXTENDED/pykickstart/pykickstart.spec [hunk 5]
```diff
@@ -0,0 +1,804 @@
...
+Release:        2%{?dist}
+License:        GPLv2 AND MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://fedoraproject.org/wiki/pykickstart
+# This is a Red Hat maintained package which is specific to
+# our distribution.  Thus the source is only available from
+# within this srpm.
+Source0:        https://github.com/pykickstart/%{name}/archive/refs/tags/r%{version}.tar.gz#/%{name}-%{version}.tar.gz
+Source1:        %{name}-%{version}.tar.gz.asc
```
[the remark this answers is not in the document]
Comment (suresh-thelkar) [reply SPECS-EXTENDED/pykickstart/pykickstart.spec]: thanks. I have removed these signatures files now.

File: SPECS-EXTENDED/pykickstart/pykickstart.spec (hunk 1 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/pykickstart/pykickstart.spec]: I have modified the link to get the source from GitHub but forgot to remove the comment. I have now deleted it.

File: SPECS-EXTENDED/pykickstart/pykickstart.spec (hunk 3 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/pykickstart/pykickstart.spec]: Done

**Comment by suresh-thelkar** [comment]
@mfrw, Thanks for the review. I have addressed your comments. Please take a look at the latest commit.

File: SPECS-EXTENDED/pykickstart/pykickstart.signatures.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+ "Signatures": {
+  "pykickstart-3.36.tar.gz": "2721fe2cdc3ae6e485f2a3344c555d6ed505b412d3dd19c1d00492a106e60b98",
+  "pykickstart-3.36.tar.gz.asc": "ef1ff421210a110dd1e11d50d0770c9333b0b51959d30507a7a50fb3bed47097"
```
Comment (mfrw) [inline SPECS-EXTENDED/pykickstart/pykickstart.signatures.json:4]: This line is a vestege of our previous removals :)

Comment (suresh-thelkar) [reply SPECS-EXTENDED/pykickstart/pykickstart.signatures.json]: I have removed this as well. Please check the latest commit.

File: SPECS-EXTENDED/pykickstart/pykickstart.spec [hunk 7]
```diff
@@ -0,0 +1,798 @@
+# Disable tests by default because they fail to run inside mock builds
+# at the moment, but can run locally.  To build and run tests, do:
+#     rpmbuild -ba --with runtests pykickstart.spec
+%bcond_with runtests
```
Comment (mfrw) [inline SPECS-EXTENDED/pykickstart/pykickstart.spec:4]:
```suggestion
```

**Review by mfrw — APPROVED** [review]
Final nitpick :)

File: SPECS-EXTENDED/pykickstart/pykickstart.spec (hunk 7 above)
Comment (suresh-thelkar) [reply SPECS-EXTENDED/pykickstart/pykickstart.spec]: thanks. Removed the above macro.

**Comment by suresh-thelkar** [comment]
I have once again ran buddy builds. Both of them are passing along with the tests. 
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=325035&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=325036&view=results)

**Pull request merged**